src/components/SingleArticle.jsx

```diff
@@ -24,7 +25,7 @@ export default function SingleArticle() {
           setCommentsLoading(false);
         });
       });
-  }, []);
+  }, [commentsByArticleId]);
```

Comment:
This is causing an infinite loop in your use effect. commentsByArticleId is set within the useEffect so the useEffect triggers itself. You will see this if you open the Network tab of the inspect in your browser.